src/state/api.js

```diff
@@ -54,3 +54,32 @@ export async function getProbeData(params, token) {
   });
   return data;
 }
+
+export async function getSearchResults(queryString, resultsLimit) {
+  const getFormattedSearchURL = (str, product = 'desktop') => {
+    const URLResult = new URL('__BASE_SEARCH_DOMAIN__');
+    const params = new URLSearchParams();
+    const queryOptions = [];
+
+    queryOptions.push(`name.eq.${str}`);
+    queryOptions.push(`index.phfts(english).${str}`);
+    queryOptions.push(`description.phfts(english).${str}`);
+    queryOptions.push(`name.ilike.*${str}*`);
+
+    params.set('limit', resultsLimit);
+    params.set('select', 'name,definition,type');
+    params.set('product', `eq.${product}`);
+    params.set('type', 'neq.event');
+    params.set('or', `(${queryOptions.join(',')})`);
+
+    URLResult.pathname = 'probes'; // hint: change this to test 404 error case
+    URLResult.search = params;
+
+    return URLResult;
+  };
+
+  return await fetch(getFormattedSearchURL(queryString)).then(async (r) => {
```

Comment:
I don't think the `await` keyword is necessary here, since I would presume that whatever utilizes the return value of this function will be awaiting anyway

```suggestion
  return fetch(getFormattedSearchURL(queryString)).then(async (r) => {
```

Reply:
👍